packages/wrangler/src/pages/build-env.ts

```diff
@@ -9,17 +12,37 @@ import type {
 export type PagesBuildEnvArgs = StrictYargsOptionsToInterface<typeof Options>;
 
 export function Options(yargs: CommonYargsArgv) {
-	return yargs;
+	return yargs.positional("projectDir", {
+		type: "string",
+		description: "The location of the Pages project",
+	});
 }
 
 export const Handler = async (args: PagesBuildEnvArgs) => {
-	const config = readConfig(undefined, {
+	if (!args.projectDir) {
+		throw new FatalError("No Pages project location specified");
+	}
+	const configPath = path.resolve(args.projectDir, "wrangler.toml");
+	// Fail early if the config file doesn't exist
+	try {
+		await stat(configPath);
+	} catch {
+		throw new FatalError(
+			"No Pages config file found",
+			EXIT_CODE_NO_CONFIG_FOUND
+		);
+	}
+	const config = readConfig(path.resolve(args.projectDir, "wrangler.toml"), {
 		...args,
 		// eslint-disable-next-line turbo/no-undeclared-env-vars
 		env: process.env.PAGES_ENVIRONMENT,
 	});
+	// Fail if the config file exists but isn't valid for Pages
 	if (!config.pages_build_output_dir) {
```

Comment:
It feels like this would be cleaner in a helper that made it clear we are just checking whether it is a valid pages config, which would avoid the need for a comment explaining this.

```suggestion
	if (!isPagesConfig(config)) {
```

Reply:
I agree. We are using this check in multiple places, so having a helper would make thing tidier and more maintainable on the long term. I remember adding this helper in one of my PRs, but not sure if it isn't in the `pages dev` PR which didn't land yet...

Reply:
oh, found it: https://github.com/cloudflare/workers-sdk/blob/main/packages/wrangler/src/config/validation.ts#L45-L47

Reply:
it prolly needs some adjusting, since rn it take a raw config object as param.

Reply:
Changed to using `isPagesConfig()`